Updated Contributing Guidelines

outoftardis opened the pull request:
I've restructured your guidelines a bit for them to be easier to scan & use (hopefully). And I also corrected wording in a few places.

tatianamac commented on CONTRIBUTING.md:
````diff
@@ -2,44 +2,58 @@
-```
+  Every definition _needs_ to have this meta information: `title`, `slug`, `defined`, and `speech`. Let's discuss what each of these mean:
+  
+  - `title` is easy. In my case it's «Obsessive Compulsive Disorder». 
````
Let's avoid the word 'easy' here. In general I'd like to ensure we don't use language that can discourage beginners and/or neurodivergent folks.

I'd propose instead something like `title` typically is the word or subject you're seeking to define.

outoftardis replied:
done

tatianamac commented on CONTRIBUTING.md:
```diff
@@ -2,44 +2,58 @@
+  
+  - `title` is easy. In my case it's «Obsessive Compulsive Disorder». 
+  
+  - `slug` is a string that we'll use to link to this word. It should be a URL-friendly, hyphenated version of the title. In my case: `obsessive-compulsive-disorder`. 
```
@ovlb Should we here suggest that it should be lowercase?

In the future we'll need to consider abbreviations and parsing; should we suggest here that it's the "full title?"

ovlb replied:
Full title and lower case make sense to be added here.

outoftardis replied:
@ovlb @tatianamac is it always full title though? In the example the title is _"Obsessive Compulsive Disorder (OCD)"_, but the slug is _obsessive-compulsive-disorder_ without the _ocd_ part.

outoftardis replied:
I've made these changes, but the question about full title still stands

ovlb replied:
Good point. I lean towards omitting abbreviations in the slug, since they are additional information to the actual title of the word. But that’s personal preference, I guess.

tatianamac replied:
I think to be explicit we should say something like (feel free to edit it for clarity, ofc):

"Please include the full phrase but omit any abbreviations or initialisms that are included in the `title`. For example, we omitted OCD."

tatianamac commented on CONTRIBUTING.md:
```diff
@@ -2,44 +2,58 @@
 Hi! Nice to have you with us. This file will walk you through the steps you need to take if you want to contribute a word or some code to the project. Let’s go.
 
-Actually – before we start – a quick word of caution: Unfortunately the contribution process involves some technical knowledge. We aware that this poses a problem for most people and are discussing how a more inclusive contribution process could look like.
+Actually – before we start – a quick word of caution: unfortunately, the contribution process involves some technical knowledge. We are aware that this poses a problem for most people and are discussing how a more inclusive contribution process could look like.
```
Remove "actually" given its very condescending connotation.

@ovlb I wonder if we can be a bit more explicit here about what technical knowledge the contribution process involves.

From what I know it's mostly understanding:
- Forking a repo
- Understanding of pull requests
- Writing markdown (though anyone who is good at copying templates could accomplish it without fluency in markdown)

Am I missing anything else?

Mostly I want this to come across as encouraging rather than discouraging. Right now it reads to me as "welp, hope you know technical stuff.

I might suggest reframing this as:

```
The contribution process currently involves some working knowledge of Github and writing markdown. We understand that this adds a barrier to entry for folks who haven't worked with either of those before. Long term, we hope to provide a solution that can minimise these sorts of barriers.

We also want encourage you to dive into Github and markdown even if you haven't before. The contributing team is always here to help. No question is "too simple" or "too easy." We'd rather you ask as someone else might want to know, too!

Below are two ways you can get started.

ovlb replied:
That’s about it! I like your reframing here. Much better approach then the one I started. 

I had already linked to the article by Chaser Pettit about this workflow. Maybe we the link into this section to give it more context? Or name it again? 

We should also link to the development documentation page as mentioned in #32.

outoftardis replied:
@tatianamac what would be the best way to contact the contributing team? we encourage folks to ask questions, how should they do this?

tatianamac replied:
I think the best way is to either email [selfdefined@tatianamac.com ](mailto:selfdefined@tatianamac.com)(which I need to work out as being its own selfdefined.app email soon) or to @ or DM [@selfdefinedapp](https://www.twitter.com/SelfDefinedApp) on Twitter.

tatianamac commented:
@outoftardis Thank you so much for your amazing contribution! You've really helped to carry this documentation forward and along. I left a few notes that hopefully add a tone of friendliness and beginner-encouragement.

ovlb commented:
Just to be clear: the general tone and voice of this document has been introduced by me.

outoftardis commented:
Thank you for your comments, I'll get back to it the moment I have a spare second!

outoftardis commented:
What I did apart from addressing your comments:
- added a link to your documentation (https://www.selfdefined.app/documentation/)
- added a link to good first issues (might be helpful)
- added a link to markdown website where they explain the basics (if you know any good markdown tutorials or cheat sheets, we can add links to those as well)
- switched from first person singular to plural

tatianamac commented on CONTRIBUTING.md:
```diff
@@ -2,44 +2,58 @@
 ### Version 1: Issues
 
-If you know a word that should be defined, but – for whatever reason – don’t feel confident writing the definition yourself (or maybe you disagree with an existing defintion), please open an issue. We can then collectively figure out how to best get this word defined.
+If you know a word that should be defined, but – for whatever reason – don’t feel confident writing the definition yourself (or maybe you disagree with an existing defintion), please open an [issue](https://github.com/tatianamac/selfdefined/issues/new). We can then collectively figure out how to best get this word defined.
```
This is preference but could we please ensure all emdashes do not have spaces around them 🙈

outoftardis replied:
Oh wow, I've just learned something new and counter-intuitive about English punctuation! I'll remove them, sure.

ovlb replied:
If you want to spend to much time on punctuation you would a thin space on either side of the em dash to avoid that the dash feels to close to the characters (m—m vs m — m). While that’s very valuable trivia to impress … mostly nobody with, it is completely useless knowledge in everyday life. 

Actually, looking closer at it, @tatianamac, the spaces in use here are en dashes (–) and as such need to be spaced. Or changed to em dashes (—).

Excuse us, Ekaterina, typographers can be incredibly annoying. 🙈

outoftardis replied:
I did change them to em dashes.

@ovlb It's fine. If the text would've been in Russian, you'd need to put spaces there. Makes you think how arbitrary all these rules are.

ovlb replied:
In German you would use en dashes with a normal space. Fun, fun, fun.

tatianamac commented on CONTRIBUTING.md:
```diff
@@ -2,44 +2,58 @@
+- Step 1: Create a file named `word.md` in in the [definitions folder](11ty/definitions/) where we store all definitions.
 
-All definitions consist of two parts: Front Matter and content. Front Matter is technical bubble for some structured data about your content. [You can find the complete documentation on our website](https://www.selfdefined.app/documentation/front-matter/). Here’s a quick run down:
+  `word` should be the name of the word you are defining. Not `word`. Except you are defining «word». In this case it should be `word`. 
```
I think this is funny but I wonder upon second read if it could be kind of confusing?

ovlb replied:
That’s my personal preference for documentation that does not take itself to seriously shining through. 🙈

Agree on the point that it might be confusing, though.

outoftardis replied:
I liked the humor as well, but it might confuse someone, that's true. I rewrote this paragraph.

tatianamac commented on CONTRIBUTING.md:
```diff
@@ -1,44 +1,63 @@
 ### Version 1: Issues
 
-If you know a word that should be defined, but – for whatever reason – don’t feel confident writing the definition yourself (or maybe you disagree with an existing defintion), please open an [issue](https://github.com/tatianamac/selfdefined/issues/new). We can then collectively figure out how to best get this word defined.
+If you know a word that should be defined, but – for whatever reason – don’t feel confident writing the definition yourself (or maybe you disagree with an existing defintion), please [open an issue](https://github.com/tatianamac/selfdefined/issues/new). We can then collectively figure out how to best get this word defined.
```
remove spaces around emdashes please :)

(Reminds me I need to open an issue for house style guide 😅)

tatianamac commented on CONTRIBUTING.md:
```diff
@@ -1,44 +1,63 @@
 #### Working on Words
 
-After you’ve forked the repo, you will need to write a definition. Good times. Hopefully. Always remember, if you are unsure, you can first open an issue or create a PR, include «WIP» in the title, and ask for early feedback.
+After you’ve forked the repo, you will need to write a definition. Good times. Hopefully. Always remember that if you are unsure, you can first open an issue or create a PR, include «WIP» in the title («WIP» stands for «work in progress»), and ask for early feedback.
```
Can we please be more explicit here and add an example? 

"For example, your issue might be called: _[WIP] «Propose definition for spirit animal»_"

outoftardis replied:
done

tatianamac commented on CONTRIBUTING.md:
```diff
@@ -1,44 +1,63 @@
   `word` should be the name of the word you are defining. Not `word`. Except you are defining «word». In this case it should be `word`. 
 
-  For this tutorial I will take the word «Obsessive Compulsive Disorder», as it allows me to show some of the features. That means I would create the file `obsessive-compulsive-disorder.md`. As it is already defined, you can have a look at the [finished definition](11ty/definitions/obsessive-compulsive-disorder.md).
+  For this tutorial we will take the word «Obsessive Compulsive Disorder», as it allows us to show some of the features. That means we would create the file named `obsessive-compulsive-disorder.md`. As it is already defined, you can have a look at the [finished definition of «Obsessive Compulsive Disorder»](11ty/definitions/obsessive-compulsive-disorder.md).
```
I think we should put "As it is..." onward into a parenthetical ()

outoftardis replied:
done

tatianamac commented on CONTRIBUTING.md:
```diff
@@ -1,44 +1,63 @@
+  - `title` is a word or a phrase you are defining. In this example it's «Obsessive Compulsive Disorder». 
   
-  - `slug` is a string that we'll use to link to this word. It should be a URL-friendly, hyphenated version of the title. In my case: `obsessive-compulsive-disorder`. 
+  - `slug` is a string that we'll use to link to this word. It should be a URL-friendly, all-lowercase hyphenated version of the full title. In our example it's `obsessive-compulsive-disorder`. 
```
This area in particular is SO MUCH CLEARER!!! 💙

tatianamac commented on CONTRIBUTING.md:
```diff
@@ -1,44 +1,63 @@
+- [Basic Markdown Syntax](https://www.markdownguide.org/basic-syntax/)
+- [Good First Issues](https://github.com/tatianamac/selfdefined/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22good+first+issue%22+)
+
+Please remember that the contributing team is always here to help. No question is "too simple" or "too easy". We'd rather you ask, as someone else might want to know, too!
```
I commented this on the previous commit (sometimes I get lost in multiple comments 😬), but we can add maybe a "Where to Get Help" section broken out? If you think that's overkill for right now, that's fine. I wrote a bit in case helpful:

> Where to Get Help
> If you've already tried reading through our [documentation](https://www.selfdefined.app/documentation/) and are stuck, we're here to help.
> 
> You can contact us/ask questions by reaching out to us on Twitter at [@SelfDefinedApp](https://www.twitter.com/selfdefinedapp).
> You can file an issue if you think some information is missing and/or if it would be helpful.

outoftardis replied:
I'll add it, but I also think it is a good practice to put "get help" info into README as well.
